### DISCUSSION TITLE
Same image appears for two components

### DISCUSSION THREAD

**Discussion by mflorindo** [alpinejs/alpine#2756] [1. Help]
I have two components:

 <x-image-card title="Image" imageCircle='true' wire:model="imageFile"  imagePreview="{{ $imageURL }}"></x-image-card>
  <x-image-card title="Logo" imageCircle='true' wire:model="logoFile"   imagePreview="{{ $logoURL }}"> </x-image-card>

  component:

  <div x-data="init()">

    <div class="card" :class="{ 'imageChange': imageChange }">
        <h4 class="text-center" :class="{ 'imageChange': imageChange }">{{ $title }}</h4>
        <div class="container">
            <div class="row">
                <div class="col-12">
                    <img @click="$refs.image.click()" style="cursor: pointer;"
                        class="card-img-top img-thumbnail mb-2 `${imageCircle}`" x-bind:src="imagePreview">

                </div>

                <div class="col-12 d-none">

                    <input type="file" x-ref="image" {{ $attributes }} @change="
                        const reader = new FileReader()
                        reader.onload = (event) => {
                            imagePreview = event.target.result ====== IT LOOKS LIKE THE PROBLEM IS IN THIS LINE.

                        }
                        reader.readAsDataURL($refs.image.files[0])
                        imageChange = true

                    ">
                </div>
            </div>

        </div>
          ........              
    </div>

    <script>
        function init() {
            return {
                imagePreview: '{{ $imagePreview }}',
                imageCircle: '{{ isset($imageCircle) ? 'rounded-circle' : 'rounded' }}',
                imageChange: false,

            }
        }
    </script>

</div>

The image of the last component appears on both when starting the page. It looks like it defaults to the imagePreview of the latter. If I put 4 components, again, the last one will be the default.

Any idea?

**Comment by SimoTod** [comment]
Where is your x-data? If you put that on a shared ancestor tag, they'll have the same data context.

**Reply by mflorindo** [reply]
Hello Simone!

in the address below I show the call of the two components and the complete component (with x-data)

https://pastebin.com/Qu41pXr0

**Reply by mflorindo** [reply]
take the last image and copy it to the previous ones.

![image](https://user-images.githubusercontent.com/4173666/158481236-0b0bf5cb-1d29-496e-b4fd-445c7e385c3e.png)

**Reply by SimoTod** [reply]
I don't have much experience with blade and livewire so I can't help much. Nothing wrong stands out from a quick glance so I assume something weird goes on with the way the page is rendered.

Try to have a look at the generated HTML to see if it's what you were expecting
